satpy/utils.py

```diff
@@ -631,6 +633,112 @@ def _get_pytroll_chunk_size():
         return None
 
 
+def chunks_by_resolution(
+        input_shape: tuple[int, ...],
```

Comment:
Not sure if this is possible, but would it make sense to work on xarray DataArrays in order to be able to pick just the x and y dimensions?

Reply:
No because we'd have to have a chunked dask array to give to the DataArray to be able to use the DataArray's dimensions (chicken/egg problem). The caller could only pass the y and x dimensions as its `input_shape` and then use the resulting chunks accordingly. I could update MODIS to do this now that I think of it. Right now it just works though too.